internal/events/loop.go

```diff
@@ -37,37 +43,32 @@ func NewEventLoop(
 	preparer FirstEventBatchPreparer,
 ) *EventLoop {
 	return &EventLoop{
-		eventCh:  eventCh,
-		logger:   logger,
-		handler:  handler,
-		preparer: preparer,
+		eventCh:      eventCh,
+		logger:       logger,
+		handler:      handler,
+		preparer:     preparer,
+		currentBatch: make(EventBatch, 0),
+		nextBatch:    make(EventBatch, 0),
 	}
 }
 
 // Start starts the EventLoop.
 // This method will block until the EventLoop stops, which will happen after the ctx is closed.
 func (el *EventLoop) Start(ctx context.Context) error {
-	// The current batch.
-	var batch EventBatch
 	// handling tells if any batch is currently being handled.
 	var handling bool
 	// handlingDone is used to signal the completion of handling a batch.
 	handlingDone := make(chan struct{})
 
-	handleAndResetBatch := func() {
+	handleBatch := func() {
```

Comment:
Since this can be run separately in one case, what do you think about two functions?

Call handleBatch on line 92.

Then also tie the operations with a swap & run.

```
func swapAndRun() {
  swap()
  handle()
  handling = true
}
```

Reply:
Done